packages/ember-utils/lib/guid.js

```diff
@@ -119,54 +98,33 @@ export function generateGuid(obj, prefix = GUID_PREFIX) {
   @param {Object} obj any object, string, number, Element, or primitive
   @return {String} the unique guid for this instance.
 */
-export function guidFor(obj) {
-  // special cases where we don't want to add a key to object
-  if (obj === undefined) {
-    return '(undefined)';
-  }
-
-  if (obj === null) {
-    return '(null)';
-  }
-
-  let type = typeof obj;
-  if ((type === 'object' || type === 'function') && obj[GUID_KEY]) {
-    return obj[GUID_KEY];
-  }
-
-  let ret;
-  // Don't allow prototype changes to String etc. to change the guidFor
-  switch (type) {
-    case 'number':
-      ret = numberCache[obj];
+export function guidFor(value) {
+  let guid;
 
-      if (!ret) {
-        ret = numberCache[obj] = `nu${obj}`;
-      }
-
-      return ret;
-
-    case 'string':
-      ret = stringCache[obj];
-
-      if (!ret) {
-        ret = stringCache[obj] = `st${uuid()}`;
-      }
-
-      return ret;
+  if (isObject(value)) {
+    guid = OBJECT_GUIDS.get(value);
 
-    case 'boolean':
-      return obj ? '(true)' : '(false)';
-
-    default:
-      if (obj === Object) {
-        return '(Object)';
-      }
-
-      if (obj === Array) {
-        return '(Array)';
+    if (guid === undefined) {
+      guid = GUID_PREFIX + uuid();
+      OBJECT_GUIDS.set(value, guid);
+    }
+  } else {
+    guid = NON_OBJECT_GUIDS.get(value);
+
+    if (guid === undefined) {
+      if (typeof value === 'string') {
+        guid = 'st' + uuid();
+      } else if (typeof value === 'number') {
```

Comment:
Not terribly significant, but I think I'd use a local variable here to avoid calling `typeof value` so many times in a row...

Reply:
Done.